### PR TITLE
feat: add pagination to projects page

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1385

- add Pagination UI component with custom styling
- add route pagination to `/projects/:page`
- display 6 project cards by page on `/projects/:page`
- update fetching logic to only fetch metadatas for the current page
- fix a rerender issue caused by an effect
- split projects normalization to separate metadata and orders infos 
- change projects routes
  - `/projects/:id` -> `/project/:id` 
  - `/projects` -> `/projects/:page`

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://deploy-preview-1668--regen-registry.netlify.app/projects/1

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by flagrede** [comment]
@erikalogie You can test the paginated projects page here: https://deploy-preview-1668--regen-registry.netlify.app/projects/1

File: web-registry/src/pages/Projects/utils/sortProjects.ts [hunk 1]
```diff
@@ -1,28 +1,5 @@
...
-    useState<ProjectWithOrderData[]>(projects);
-
-  useEffect(() => {
-    const projectsCopy = [...projects]; // to make sure render picks up change
-    const _sortedProjects = sortProjects(projectsCopy, sort);
-    setSortedProjects(_sortedProjects);
-  }, [projects, sort]);
-
-  return sortedProjects;
-};
```
Comment (flagrede) [inline web-registry/src/pages/Projects/utils/sortProjects.ts:24]: This was the effect that was causing rerender. We're now using sort functions directly instead.

**Review by mhagel — APPROVED** [review]
This is great!

**Review by victordelval — APPROVED** [review]
Nice work!

**Review by blushi — COMMENTED** [review]
can we update _redirects to map projects/{id} to project/{id} for existing projects?

**Comment by flagrede** [comment]
> can we update _redirects to map projects/{id} to project/{id} for existing projects?

unfortunately, we can't because projects/{id} would be the same that projects/{page}, there's no way to differentiate the two from a URL standpoint.

**Comment by blushi** [comment]
> > can we update _redirects to map projects/{id} to project/{id} for existing projects?
> 
> unfortunately, we can't because projects/{id} would be the same that projects/{page}, there's no way to differentiate the two from a URL standpoint.

I meant explicitly using the handle/ids of the existing projects
eg projects/wilmot -> project/wilmot

**Comment by flagrede** [comment]
> > > can we update _redirects to map projects/{id} to project/{id} for existing projects?
> > 
> > 
> > unfortunately, we can't because projects/{id} would be the same that projects/{page}, there's no way to differentiate the two from a URL standpoint.
> 
> I meant explicitly using the handle/ids of the existing projects (on prod) eg projects/wilmot -> project/wilmot

Oh ok that we can do, will add that.

File: web-registry/public/_redirects [hunk 2]
```diff
@@ -1,2 +1,10 @@
-https://registry.regen.network/*  https://app.regen.network/:splat 301!
-/*    /index.html   200
+https://registry.regen.network/_ https://app.regen.network/:splat 301!
+https://app.regen.network/projects/C02-002 https://app.regen.network/project/C02-002 301!
```
Comment (blushi) [inline web-registry/public/_redirects:2]: wouldn't that work to just use relative paths?
we might also want to to add off-chain projects which might have been shared publicly in the past:
/wilmot, /cavan-station, /woodburn and /pillango

File: web-registry/public/_redirects [hunk 3]
```diff
@@ -1,2 +1,10 @@
-https://registry.regen.network/*  https://app.regen.network/:splat 301!
-/*    /index.html   200
+https://registry.regen.network/_ https://app.regen.network/:splat 301!
```
[the remark this answers is not in the document]
Comment (flagrede) [reply web-registry/public/_redirects:1]: Had no idea how this end up here, so I tried to revert the `*`, and when I tried it goes back to `_`. I guess it must have been prettier on-save doing...

**Review by blushi — APPROVED** [review]

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**